unified_planning/plans/sequential_plan.py

```diff
@@ -0,0 +1,104 @@
+        # graph stores the information gathered through the process
+        graph = nx.DiGraph()
+        for action_instance in self.actions:
+            grounded_action = cast(InstantaneousAction, up.plans.plan.ground_action_instance(action_instance, subs, simp))
+            graph.add_node(action_instance)
+
+            # required_fluents contains all the fluents that this action_instance "reads"
+            required_fluents: Set[FNode] = set()
+            # add free vars of preconditions
+            for prec in grounded_action.preconditions:
+                required_fluents = required_fluents.union(fve.get(prec))
+            # add in the required_fluents all the free fluents this action instance deals with
+            for eff in grounded_action.effects:
+                required_fluents = required_fluents.union(fve.get(eff.condition))
+                required_fluents = required_fluents.union(fve.get(eff.fluent))
+                required_fluents = required_fluents.union(fve.get(eff.value))
+
+            # for every required fluent, add this action instance to the list of action instances that requires this fluent
+            # and order the current action instance after the last modifier of the fluent
+            for required_fluent in required_fluents:
+                action_instance_list = all_required.get(required_fluent, None)
+                if action_instance_list is None:
+                    all_required[required_fluent] = [action_instance]
+                else:
+                    action_instance_list.append(action_instance)
+                required_fluent_last_modifier = last_modifier.get(required_fluent, None)
+                if required_fluent_last_modifier is not None:
+                    assert required_fluent_last_modifier != action_instance # sanity
+                    graph.add_edge(required_fluent_last_modifier, action_instance)
+
+            # for every effect, set current action instance as the last modifier and the current action instance is ordered
+            # after every action instance that requires a fluent the current action instance modifies
+            for eff in grounded_action.effects:
+                assert eff.fluent.is_fluent_exp()
+                last_modifier[eff.fluent] = action_instance
+                dependent_action_instance_list = all_required.get(eff.fluent, None)
+                if dependent_action_instance_list is not None:
+                    for dependent_action_instance in dependent_action_instance_list:
+                        if dependent_action_instance != action_instance:
+                            graph.add_edge(dependent_action_instance, action_instance)
+
+        # remove redundant edges and return the up.plans.partial_order_plan.PartialOrderPlan structure.
+        assert nx.is_directed_acyclic_graph(graph)
+        return up.plans.partial_order_plan.PartialOrderPlan(nx.convert.to_dict_of_lists(nx.transitive_reduction(graph)))
```

Comment:
Make yourself a favor and create a constructor that accepts a `nx.DiGraph` for internal use